src/blocks/HighlightBlock/Component.tsx

```diff
@@ -0,0 +1,92 @@
+import { Article, Category, HighlightBlock as HighlightBlockProps } from '@/payload-types'
+import React from 'react'
+import { getPayload } from 'payload'
+import config from '@payload-config'
+import { HighlightedElement, HighlightElementsComponent } from './elements/Component'
+
+type Props = HighlightBlockProps
+
+const getProductElements = async (productArray: any[]): Promise<HighlightedElement[]> => {
```

Comment:
Let's try to avoid `any` types as much as possible.
```suggestion
const getProductElements = async (productArray: { product?: EuProduct | null | undefined | number }[]): Promise<HighlightedElement[]> => {
```